pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -230,6 +229,23 @@ func (lp *LifecycleProcessor) OnInvokeStart(startDetails *InvocationStartDetails
 		}
 		ev = event
 		lp.initFromLambdaFunctionURLEvent(event, region, account, resource)
+	case trigger.LegacyStepFunctionEvent:
+		var event events.StepFunctionEvent
+		if err := json.Unmarshal(payloadBytes, &event); err != nil {
+			log.Debugf("Failed to unmarshal %s event: %s", stepFunction, err)
+			break
+		}
+		ev = event.Payload
+		lp.initFromStepFunctionEvent(event)
+	case trigger.StepFunctionEvent:
+		var eventPayload events.StepFunctionPayload
+		if err := json.Unmarshal(payloadBytes, &eventPayload); err != nil {
+			log.Debugf("Failed to unmarshal %s event: %s", stepFunction, err)
+			break
+		}
+		sfe := events.StepFunctionEvent{Payload: eventPayload}
+		ev = eventPayload
+		lp.initFromStepFunctionEvent(sfe)
```

Comment:
Nice. I like this. This way, the event passed out of this method for both of these step function types will be the same.